scheduler/replica_scheduler.go

```diff
@@ -261,24 +260,8 @@ func (rcs *ReplicaScheduler) getDiskCandidates(nodeInfo map[string]*longhorn.Nod
 			return diskCandidates, nil
 		}
 		multiError.Append(errors)
-	case !zoneSoftAntiAffinity && nodeSoftAntiAffinity:
-		diskCandidates, errors := getDiskCandidatesFromNodes(unusedNodesInNewZones)
-		if len(diskCandidates) > 0 {
-			return diskCandidates, nil
-		}
-		multiError.Append(errors)
-		diskCandidates, errors = getDiskCandidatesFromNodes(nodesInUnusedZones)
-		if len(diskCandidates) > 0 {
-			return diskCandidates, nil
-		}
-		multiError.Append(errors)
-		diskCandidates, errors = getDiskCandidatesFromNodes(nodesWithEvictingReplicas)
```

Comment:
We can change this to `getDiskCandidatesFromNodes(filterNodesWithLessThanTwoReplicas(nodesWithEvictingReplicas))` as in the `!zoneSoftAntiAffinity && nodeSoftAntiAffinity` case. It is not legal to schedule a replica to a node with more than one (evicting) replica, because doing so would put two replicas in the same zone.